sdk/program/src/stake/state.rs

```diff
@@ -11,18 +11,51 @@ use {
         stake_history::{StakeHistory, StakeHistoryEntry},
     },
     borsh::{maybestd::io, BorshDeserialize, BorshSchema, BorshSerialize},
-    std::collections::HashSet,
+    std::{collections::HashSet, ops::Add},
 };
 
 pub type StakeActivationStatus = StakeHistoryEntry;
 
+#[repr(u8)]
+#[derive(
+    Debug,
+    Default,
+    Serialize,
+    Deserialize,
+    PartialEq,
+    Clone,
+    Copy,
+    AbiExample,
+    BorshDeserialize,
+    BorshSchema,
+    BorshSerialize,
+)]
+
+pub enum DeactivationFlag {
+    #[default]
+    Empty = 0,
+    MustFullyActivateBeforeDeactivationIsPermitted = 1,
+}
+
+impl Add for DeactivationFlag {
+    type Output = Self;
+
+    fn add(self, other: Self) -> Self {
+        if self == DeactivationFlag::Empty && other == DeactivationFlag::Empty {
+            DeactivationFlag::Empty
+        } else {
+            DeactivationFlag::MustFullyActivateBeforeDeactivationIsPermitted
+        }
+    }
+}
+
```

Comment:
can we not overload `Add` with unintuitively not-add-like behavior like this?  prefer a self-descriptive method name instead

Reply:
Done.